frontend/src/metabase/entities/containers/EntityListLoader.jsx

```diff
@@ -114,19 +117,28 @@ export default class EntityListLoader extends React.Component {
     );
   }
 
-  async fetchList(
+  maybeDebounce(f, interval, immediate) {
+    if (this.props.debounced) {
+      return _.debounce(f, interval, immediate);
+    } else {
+      return f;
+    }
+  }
+
+  fetchList = this.maybeDebounce(async (
     // $FlowFixMe: fetchList provided by @connect
     { fetchList, entityQuery, pageSize, onChangeHasMorePages },
     options?: any,
-  ) {
+  ) => {
     const result = await fetchList(entityQuery, options);
+
     if (typeof pageSize === "number" && onChangeHasMorePages) {
       onChangeHasMorePages(
         !result.payload.result || result.payload.result.length === pageSize,
       );
     }
     return result;
-  }
+  }, 250);
```

Comment:
Will this ever need to change?

Reply:
Short answer: I don't know, but I think it's okay for now.

Long answer:

If it's too big, then search won't feel "live", and I also think we could have dangling characters at the end that aren't search for. For example, if someone tended to type 4 characters per debounce interval:

```
this is a search string
|    |    |    |    |
|    |    |    |    >search "this is a search stri"
|    |    |    >search "this is a search"
|    |    >search "this is a s"
|    >search "this i"
|>search "t"
```

the "ng" at the end would never be searched for unless they hit enter.


If it's too small, we lose any benefits of debouncing.


On my computer with its middle-of-the-road performance and a variety of typing speeds 1000 and 500 were too much and 250 felt okay. But it's worth keeping in mind that we can tweak it as more people test it.

Also, if we end up needing to debounce other uses of the `EntityListLoader` and have different values for each one we could change `debounced` from a boolean to an integer that represented the interval. For now it just felt like a case of YAGNI, and I appreciated the symmetry of `debounced` with `wrapped` and `reload` (in `<Search.ListLoader wrapped reload debounced>`)

Reply:
Agreed, 250ms seems like a good, middle-of-the-road debounce num for this